src/server/identifyController.py

```diff
@@ -0,0 +1,126 @@
+import cherrypy
+import urllib
+from genshi.template import TemplateLoader
+from threading import Lock
+
+SESSION_KEY = 'username'
+
+
+def check_identify(*args, **kwargs):
+  """
+  A tool that looks in config for 'identify.require'. If found and it
+  is not None, a login is required if user is not identified already.
+  """
+  login_required = cherrypy.request.config.get('identify.require', False)
+  from_page = urllib.quote(cherrypy.request.request_line.split()[1])
```

Comment:
@vphilippon  This line could be moved to the else section before the raise redirect since it is only needed there